include/swift/Basic/InlineBitfield.h

```diff
@@ -117,71 +121,12 @@ constexpr unsigned bitmax(unsigned a, unsigned b) {
 }
 
 constexpr unsigned countBitsUsed(uint64_t arg) {
-  // Is there a C++ "std::countLeadingZeros()"?
-  return (arg & 1ull << 63 ? 63 :
-          arg & 1ull << 62 ? 62 :
-          arg & 1ull << 61 ? 61 :
-          arg & 1ull << 60 ? 60 :
-          arg & 1ull << 59 ? 59 :
-          arg & 1ull << 58 ? 58 :
-          arg & 1ull << 57 ? 57 :
-          arg & 1ull << 56 ? 56 :
-          arg & 1ull << 55 ? 55 :
-          arg & 1ull << 54 ? 54 :
-          arg & 1ull << 53 ? 53 :
-          arg & 1ull << 52 ? 52 :
-          arg & 1ull << 51 ? 51 :
-          arg & 1ull << 50 ? 50 :
-          arg & 1ull << 49 ? 49 :
-          arg & 1ull << 48 ? 48 :
-          arg & 1ull << 47 ? 47 :
-          arg & 1ull << 46 ? 46 :
-          arg & 1ull << 45 ? 45 :
-          arg & 1ull << 44 ? 44 :
-          arg & 1ull << 43 ? 43 :
-          arg & 1ull << 42 ? 42 :
-          arg & 1ull << 41 ? 41 :
-          arg & 1ull << 40 ? 40 :
-          arg & 1ull << 39 ? 39 :
-          arg & 1ull << 38 ? 38 :
-          arg & 1ull << 37 ? 37 :
-          arg & 1ull << 36 ? 36 :
-          arg & 1ull << 35 ? 35 :
-          arg & 1ull << 34 ? 34 :
-          arg & 1ull << 33 ? 33 :
-          arg & 1ull << 32 ? 32 :
-          arg & 1ull << 31 ? 31 :
-          arg & 1ull << 30 ? 30 :
-          arg & 1ull << 29 ? 29 :
-          arg & 1ull << 28 ? 28 :
-          arg & 1ull << 27 ? 27 :
-          arg & 1ull << 26 ? 26 :
-          arg & 1ull << 25 ? 25 :
-          arg & 1ull << 24 ? 24 :
-          arg & 1ull << 23 ? 23 :
-          arg & 1ull << 22 ? 22 :
-          arg & 1ull << 21 ? 21 :
-          arg & 1ull << 20 ? 20 :
-          arg & 1ull << 19 ? 19 :
-          arg & 1ull << 18 ? 18 :
-          arg & 1ull << 17 ? 17 :
-          arg & 1ull << 16 ? 16 :
-          arg & 1ull << 15 ? 15 :
-          arg & 1ull << 14 ? 14 :
-          arg & 1ull << 13 ? 13 :
-          arg & 1ull << 12 ? 12 :
-          arg & 1ull << 11 ? 11 :
-          arg & 1ull << 10 ? 10 :
-          arg & 1ull << 9 ? 9 :
-          arg & 1ull << 8 ? 8 :
-          arg & 1ull << 7 ? 7 :
-          arg & 1ull << 6 ? 6 :
-          arg & 1ull << 5 ? 5 :
-          arg & 1ull << 4 ? 4 :
-          arg & 1ull << 3 ? 3 :
-          arg & 1ull << 2 ? 2 :
-          arg & 1ull << 1 ? 1 : 0
-      ) + 1;
+// Assumes uint64_t is the same as unsigned long long
+#if defined(_MSC_VER)
+  return 64u - static_cast<unsigned>(__lzcnt64(arg));
```

Comment:
This is problematic for Windows: this is AMD64 specific and not portable to other architectures (e.g. ARM64).  Could you try something like this instead:

```suggestion
#if defined(_M_AMD64)
  return 64u - static_cast<unsigned>(__lzcnt64(arg));
#elseif  defined(_M_ARM) || defined(_M_ARM64)
  return 64u - static_cast<unsigned>(_CountLeadingZeros64(arg));
#else
# error unsupported architecture
#endif
```

This effectively prevents the 32-bit x86 builds of the toolchain, but I suppose that isn't too terrible.

Reply:
Just random question here, what about instead of `error unsupported architecture` just fallback to what is being done currently which is platform independent? Also @compnerd does llvm implementation https://github.com/llvm/llvm-project/blob/d480f968ad8b56d3ee4a6b6df5532d485b0ad01e/llvm/include/llvm/Support/MathExtras.h#L225 mentioned before could be used here? Looking through the implementation it seems to already do those checks ... but I'm just asking for curiosity about it =]

Reply:
I would actually rank using the MathExtras' implementation from LLVMSupport as the best option and keeping the current implementation as a fallback as a better option to my suggestion of preventing the build.  I was weighing the tradeoff here of code complexity (well, ugliness in this case) and the impact.  The removal would just prevent windows x86 - not the best, but not the worst and something that I could theoretically restore if there was a strong reason.  But, using the existing portable operation from LLVMSupport is certainly better.

Reply:
Tried that and it wouldn’t compile because of constexpr

Reply:
Looking through the code base usages of this function seems to be more in contexts for convenience that are probably evaluated at compile time e.g. enum declarations. Code that have runtime impact seem to already use `llvm::countLeadingZeros`, so I wonder if this kind of perf improvement is worth it? Or maybe it can be more towards make the code nicer e.g. a for loop counting bits with bit mask that looks nice and still `constexpr` ... it will be an improvement I think. This is only a suggestion, I'm not a reviewer on this, was just randomly looking some PRs and found this interesting :) 
cc @compnerd

Reply:
Ah, seems like I'm wrong, a for loop in that context can't be `constexpr`, so nevermind, this would not work